google/cloud/spanner/internal/partial_result_set_source.h

```diff
@@ -85,7 +85,7 @@ class PartialResultSetSource : public spanner::ResultSourceInterface {
   // any data in (or previously in) `rows_`. When disengaged, we have already
   // delivered data that would be replayed, so resumption is disabled until we
   // see a new token.
-  absl::optional<std::string> resume_token_ = std::string{};
+  absl::optional<std::string> resume_token_;
```

Comment:
I'm not sure how that initialization was redundant.  It seems like it would leave the member in a different state.

Reply:
https://releases.llvm.org/8.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-redundant-member-init.html#:~:text=Finds%20member%20initializations%20that%20are,if%20they%20were%20not%20present.

Does creating nothing in the string mean it is a absl::nullopt?

Reply:
The default constructor for a `std::optional` is `std::nullopt`. https://en.cppreference.com/w/cpp/utility/optional/optional

@devbww is pointing out that there is a difference between:
- an optional without a value
- an optional with a value that is the empty string.